pkg/buildpipeline/pipeline.go

```diff
@@ -572,7 +585,7 @@ func stageToTask(s Stage, pipelineIdentifier string, buildIdentifier string, nam
 		o := s.Options
 		if !equality.Semantic.DeepEqual(o.Timeout, Timeout{}) {
 			return nil, errors.New("Timeout on stage not yet supported")
-		} else if !equality.Semantic.DeepEqual(o.Retry, 0) {
+		} else if o.Retry != 0 {
```

Comment:
Not really sure why this was needed, but without it the custom workspace test fails. I don't really understand what equality.Semantic.DeepEqual is doing, so probably need to read some docs.

Reply:
`equality.Semantic.DeepEqual` is basically for `struct`/`map`/`slave` comparison - so comparing a primitive, it just looks at it funny and always returns false.

Reply:
Hmm ok, thanks! Still seems weird that all of the other tests passed though with the other code

Reply:
They never actually got to this line thanks to the if/else with timeouts before it. =) I uncovered this in #4.

Reply:
Oh duh, whoops! It only happened for my new test because that was the first test to use `StageOptions` in a non-failing test.